File: pset6/caesar.py
import cs50
import sys

# If improper arguments given return 1 and error message
if len(sys.argv) != 2:
    print("Please enter only a single argument to the program")
    exit(1)

# Requrest and store plaintext
print("plaintext:", end=" ")

plaintext = cs50.get_string()

k = sys.argv[1]
k=int(k)

ln = len(plaintext)

print("cyphertext:", end=" ")

# Check if the letter is upper case of lower case and rotate by k maintaining case
for i in range(ln):
    if ord(plaintext[i]) > 64 and ord(plaintext[i]) < 91:
        c = ord(plaintext[i]) - 65
        c = (c + k)%26
        c += 65
        print(chr(c), end="")
        
    elif ord(plaintext[i]) > 96 and ord(plaintext[i]) < 123:
        c = ord(plaintext[i]) - 97
        c = (c + k)%26
        c += 97
        print(chr(c), end="")
    else:
        print(plaintext[i], end="")

print("")
# end program returning 0
exit(0)

File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
   // verify that the input is acceptable and return an error message and 1 if
   // anything other than one command line argument is given
    while (argc != 2)
    {
        printf("Input Error\n");
        return 1;
    }
    // promt for plaintext and store in string plaintext
    printf("plaintext:");
    string plaintext = get_string();
    printf("ciphertext:");
    // create int variable k from the command line input 
    int k = atoi(argv[1]);
    // for loop goes through every part of plaintext cheching if 
    // what is at the array location i is an upper case letter, 
    // a lower case letter, or anything else
    for (int i = 0; i < strlen(plaintext); i++)
    {
        // check for upper case letters, create integer if it is an
        // upper case letter of that letter on a 0-25 scale to make modulating
        // with the k mod 26 give a 0-25 number which can be 
        // retranslated into an upper case letter by adding 65
        if (isupper(plaintext[i]))
        {
            int letter = plaintext[i] - 65;
            letter = (letter + k) % 26;
            letter = letter + 65;
            printf("%c", letter);
        }
        // check for a lower case letter and, if it is one, translate it into an
        // integer between 0-25, compute its new value with k mod 26, and
        // retranslate into a lower case number by adding 97
        else if (islower(plaintext[i]))
        {
            int letter = plaintext[i] - 97;
            letter = (letter + k) % 26;
            letter = letter + 97;
            printf("%c", letter);
        }
        // if the value of plaintext[i] is not a letter, simply print the value
        else 
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");
    return 0;
}